main.go

```diff
@@ -81,6 +82,15 @@ func main() {
 		os.Exit(exitFailure)
 	}
 
+	// Wait on SIGUSR1 if it is an init process shim, in which case
+	// container ID equals to exec ID.
+	if container == execId {
+		waitSigUsr1 := make(chan os.Signal, 1)
+		signal.Notify(waitSigUsr1, syscall.SIGUSR1)
+		<-waitSigUsr1
```

Comment:
I think we could put a timeout here, just in case we never get the `SIGUSR1`signal, if something got wrong on the runtime side.
Something like one minute (60 seconds) could be okay. I mean that we don't need an aggressive timeout here.
WDYT ?

Reply:
It might not be an invalid case to wait forever. E.g., if `kata-runtime create` is called without following up `kata-runtime start`, we need to keep the shim as long as the container is created.

Reply:
Yes I agree, but I have raised this question to get the opinion of everybody on this topic (cc @sameo @laijs @WeiZhang555 )

Reply:
@bergwolf - that scenario would be mostly (wholly?) for when the runtime is [invoked manually](https://github.com/clearcontainers/runtime/wiki/Running-standalone) right? I think standalone mode is important but since it won't be the normal way the runtime is invoked, how about we add a command-line option to handle the timeout @sboeuf mentions? That way, we can handle both scenarios (timeout and wait forever) and that CLI option could be very useful for testing purposes.

Reply:
sounds good to me @jodh-intel